.github/workflows/kserve_m2m_kind_test.yaml

```diff
@@ -0,0 +1,70 @@
+name: Deploy and test KServe with m2m auth in KinD
+on:
+  pull_request:
+    paths:
+      - contrib/kserve/**
+      - common/knative/**
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Install KinD
+        run: ./tests/gh-actions/install_kind.sh
+
+      - name: Create KinD Cluster
+        run: kind create cluster --config tests/gh-actions/kind-cluster.yaml
+
+      - name: Install kustomize
+        run: ./tests/gh-actions/install_kustomize.sh
+
+      - name: Create kubeflow namespace
+        run: kustomize build common/kubeflow-namespace/base | kubectl apply -f -
+
+      - name: Install Istio with ext auth
+        run: ./tests/gh-actions/install_istio_with_ext_auth.sh*
+
+      - name: Install cert-manager
+        run: ./tests/gh-actions/install_cert_manager.sh
+
+      - name: Install knative
+        run: ./tests/gh-actions/install_knative.sh
+
+      - name: Build & Apply manifests
+        run: ./tests/gh-actions/install_kserve.sh
+
+      - name: Create test namespace
+        run: kubectl create ns kserve-test
+
+      - name: Setup python 3.9
+        uses: actions/setup-python@v4
+        with:
+          python-version: 3.9
+
+      - name: Install test dependencies
+        run: pip install -r ./contrib/kserve/tests/requirements.txt
+
+      - name: Port forward
+        run: |
+          INGRESS_GATEWAY_SERVICE=$(kubectl get svc --namespace istio-system --selector="app=istio-ingressgateway" --output jsonpath='{.items[0].metadata.name}')
+          nohup kubectl port-forward --namespace istio-system svc/${INGRESS_GATEWAY_SERVICE} 8080:80 &
+
+      - name: Run kserve tests with m2m token from SA default/default
```

Comment:
Please add some security tests:
1. userA namespace with serviceaccounttoken  from userA
2. userA namespace with serviceaccounttoken  from userA
3. userA namespace with serviceaccounttoken  from userB
4. userB namespace with serviceaccounttoken  from userA

The last two tests should fail.

Reply:
For Notebooks and Pipelines it makes a lot of sense but I'm having doubts for kserve - currently in the test scenario there is just a new namespace create and the model is not served as a part of a kubeflow profile. Because of that there is not authorizationpolicy that controls access to the model.

I admit my experience with kserve is limited so some guidance or at least relevant links would be appreciated, specifically about listing all kserver models. Also, is it standard to deploy kserve model in Kubeflow outside of KF Profile?